source/simulator/solver_schemes.cc

```diff
@@ -908,24 +952,25 @@ namespace aspect
                       (1.0-(dcr.newton_residual_for_derivative_scaling_factor/dcr.switch_initial_residual))));
 
         assemble_and_solve_defect_correction_Stokes(dcr, use_picard);
+        relative_residual = dcr.residual/dcr.initial_residual;
 
         if (parameters.run_postprocessors_on_nonlinear_iterations)
-          postprocess ();
-
-        if (dcr.residual/dcr.initial_residual < parameters.nonlinear_tolerance)
-          break;
+        postprocess ();
       }
 
-    // Reset the Newton stabilization at the end of the timestep.
-    newton_handler->parameters.preconditioner_stabilization = starting_preconditioner_stabilization;
-    newton_handler->parameters.velocity_block_stabilization = starting_velocity_block_stabilization;
+  // Reset the Newton stabilization at the end of the timestep.
+  newton_handler->parameters.preconditioner_stabilization = starting_preconditioner_stabilization;
```

Comment:
what happened with the indentation here?

Reply:
I think you lost a closing scope brace?